featuretools/tests/entityset_tests/test_entity.py

```diff
@@ -158,3 +160,62 @@ def test_variable_types_unmodified():
                              time_index='transaction_time',
                              variable_types=variable_types)
     assert old_variable_types == variable_types
+
+
+def test_passing_strings_to_variable_types_entity_init():
+    variable_types = find_variable_types()
+    reversed_variable_types = {str(v): k for k, v in variable_types.items()}
+    reversed_variable_types['unknown variable'] = 'some unknown type string'
+
+    es = EntitySet()
+    dataframe = pd.DataFrame(columns=list(reversed_variable_types))
+    entity = Entity('reversed_variable_types', dataframe, es,
+                    variable_types=reversed_variable_types,
+                    index="<class 'featuretools.variable_types.variable.Index'>",
+                    time_index="<class 'featuretools.variable_types.variable.DatetimeTimeIndex'>",
+                    )
+    for variable in entity.variables:
+        description = variable.to_data_description()
+        _variable = deserialize.description_to_variable(description, entity=entity)
+        assert variable.__eq__(_variable)
+
+
+def test_passing_strings_to_variable_types_from_dataframe():
+    variable_types = find_variable_types()
+    reversed_variable_types = {str(v): k for k, v in variable_types.items()}
+    reversed_variable_types['unknown variable'] = 'some unknown type string'
+
+    es = EntitySet()
+    dataframe = pd.DataFrame(columns=list(reversed_variable_types))
+    es.entity_from_dataframe(
+        entity_id="reversed_variable_types",
+        dataframe=dataframe,
+        index="<class 'featuretools.variable_types.variable.Index'>",
+        time_index="<class 'featuretools.variable_types.variable.DatetimeTimeIndex'>",
+        variable_types=reversed_variable_types)
+
+    entity = es["reversed_variable_types"]
+    for variable in entity.variables:
+        variable_class = variable.__class__
+        assert variable_class == variable_types[variable_class.type_string]
+
+
+def test_passing_strings_to_variable_types_dfs():
+    variable_types = find_variable_types()
+    teams = pd.DataFrame({
+        'id': range(3),
+        'name': ['Breakers', 'Spirit', 'Thorns']
+    })
+    games = pd.DataFrame({
+        'id': range(5),
+        'home_team_id': [2, 2, 1, 0, 1],
+        'away_team_id': [1, 0, 2, 1, 0],
+        'home_team_score': [3, 0, 1, 0, 4],
+        'away_team_score': [2, 1, 2, 0, 0]
+    })
+    entities = {'teams': (teams, 'id', None, {'name': 'text'}), 'games': (games, 'id')}
+    relationships = [('teams', 'id', 'games', 'home_team_id')]
+
+    _, features = ft.dfs(entities, relationships, target_entity="teams")
+    name_class = features[0].entity.variables[1].__class__
```

Comment:
An entity's variables can be looked up by name directly from the entity object

```
features[0].entity['name'].__class__
```

Small change but I think it makes the test a little easier to follow